### ISSUE TITLE
Quieten output on success

### ISSUE THREAD

**Issue by Mahoney** [autonomousapps/dependency-analysis-android-gradle-plugin#381]
At the moment the plugin logs at `quiet` level: https://docs.gradle.org/current/userguide/logging.html

This is by gradle standards very noisy on success - my output for a `./gradlew build` looks like this:
```
Type-safe dependency accessors is an incubating feature.

> Task :buildHealth
Advice for root project
Looking good! No changes needed

Advice for project :app
Looking good! No changes needed

Advice for project :auction-api
Looking good! No changes needed

Advice for project :auction-sol
Looking good! No changes needed

Advice for project :auction-stub
Looking good! No changes needed

Advice for project :auction-xmpp
Looking good! No changes needed

Advice for project :auction-xmpp-integration-tests
Looking good! No changes needed

Advice for project :clioptions
Looking good! No changes needed

Advice for project :core
Looking good! No changes needed

Advice for project :end-to-end-tests
Looking good! No changes needed

Advice for project :kotlinlangext
Looking good! No changes needed

Advice for project :retry
Looking good! No changes needed

Advice for project :testcontainers
Looking good! No changes needed

Advice for project :testlauncher
Looking good! No changes needed

Advice for project :ui-api
Looking good! No changes needed

Advice for project :ui-stub
Looking good! No changes needed

Advice for project :ui-swing
Looking good! No changes needed


BUILD SUCCESSFUL in 24s
421 actionable tasks: 19 executed, 402 up-to-date
```

**Describe the solution you'd like**
I don't think a check plugin reporting that the check passed shouldn't be logging at that level.

I think it should be logging at `info` if no advice is given. Maybe `warn` if advice was given. Unfortunately the logic to decide whether advice was given is mixed up in the logic to build the message, so it's not a trivial PR to write, but if you suggest you'd accept it I'll have a go at the refactor.

**Describe alternatives you've considered**
I tried setting `systemProp.dependency.analysis.silent=true` in `gradle.properties` and just running `./gradlew -Ddependency.analysis.silent=true build`, but neither silenced the output. If the flags in `Flags.kt` were gradle properties rather than system properties then they could be configured in `gradle.properties`.

**Additional context**

**Comment by autonomousapps** [comment]
Hi @Mahoney , sorry about the delayed response. This slipped under my radar.

I agree with you that the plugin is too verbose, especially in the "all clear" case you describe here. I would be willing to accept a PR if you want to take a shot at it. Otherwise I can't say when I'd get to it.

Regarding those system property flags, I do not use them consistently :) They're more of a diagnostic tool for me, or an unfinished refactor in some cases. Btw, you can set system properties in `gradle.properties`.

**Comment by vlsi** [comment]
+1 for removing "Looking good! No changes needed" output.
In other words, a successful task should produce empty output.

On top of that, there should be extra indentation after "Advice for project".

Here's a sample from 0.73.0:

```
  Advice for project :cassandra
  Transitively used dependencies that should be declared directly as indicated:
  - api("org.checkerframework:checker-qual:3.10.0")

  Advice for project :core
  Existing dependencies which should be modified to be as indicated:
  - testImplementation("org.apache.commons:commons-dbcp2:2.6.0") (was implementation)

  Plugin advice:
  - java-library: this project has both java-library and org.jetbrains.kotlin.jvm applied, which is redundant. You can remove java-library

  Advice for project :server
  Existing dependencies which should be modified to be as indicated:
  - implementation(project(":linq4j")) (was api)
```

Note how `Plugin advice` looks very similar to `Advice for project`.

The following would be much better:
```
  Advice for project :cassandra
    Transitively used dependencies that should be declared directly as indicated:
    - api("org.checkerframework:checker-qual:3.10.0")

  Advice for project :core
    Existing dependencies which should be modified to be as indicated:
    - testImplementation("org.apache.commons:commons-dbcp2:2.6.0") (was implementation)

    Plugin advice:
    - java-library: this project has both java-library and org.jetbrains.kotlin.jvm applied, which is redundant. You can remove java-library

  Advice for project :server
    Existing dependencies which should be modified to be as indicated:
    - implementation(project(":linq4j")) (was api)
```

**Comment by autonomousapps** [comment]
@Mahoney @vlsi please let me know if either of you would like to take a look at this. I'm willing to provide advice if necessary. I've decided this is worth fixing for 1.0, so it's going to be done sooner or later, and I want to encourage external contributions if someone is interested.

**Comment by vlsi** [comment]
Ah, you know what?
I even cloned d-a-a-g-p, and then I looked around the sources, I noticed `antlr` jar, I even tried replacing `committed under git` jars with `shaded during the build` ones, and I forgot to augment the output :)

Just in case, I see you are affected by https://github.com/gradle/gradle/issues/820. It causes 14 megabyte antlr.jar while 0.5M is just enough.
I wonder what is your take on third-party jars in git, and, frankly speaking, I would love to get them removed even from historical commits with https://github.com/vlsi/bfg-repo-cleaner

Going back to the topic, I might indeed have some time to cleanup the output.
